### ISSUE TITLE
High fidelity mockup of the application inbox view

### ISSUE THREAD

**Issue by gcapiel** [seattle-uat/civiform#492]
High fidelity mockup of the application inbox view. Wireframes [here](https://miro.com/app/board/o9J_lSQyXFI=/?moveToWidget=3074457355718069753&cot=14), styles & design patterns can be pulled from these [mockups](https://miro.com/app/board/o9J_lSQyXFI=/?moveToWidget=3074457355718069770&cot=14).

**Comment by neel-a-mehta** [comment]
High fidelity mockup of the application inbox view (for program admins who are receiving applications and approving them). Wireframes [here](https://miro.com/app/board/o9J_lSQyXFI=/?moveToWidget=3074457355718069753&cot=14), styles & design patterns can be pulled from these [mockups](https://miro.com/app/board/o9J_lSQyXFI=/?moveToWidget=3074457355718069770&cot=14).

**Comment by neel-a-mehta** [comment]
Copying my message about this over from Slack...

> we have some low-fidelity wireframes for the Application Inbox view [here](https://miro.com/app/board/o9J_lXULxwE=/?moveToWidget=3074457354414619493&cot=14). @HansEngebretsen  might know if we've done anything higher-fi.

**Comment by neel-a-mehta** [comment]
Here's what it looks like right now. Please make some better mockups

![Uploading image.png…]()

**Comment by neel-a-mehta** [comment]
- Support potentially thousands of applications
- Paging
- Two-panel layout with full details about the application on the right, summaries on the left
- Visit <https://seattle.civiform.com> to try applying as a guest
- Enable some kind of filter view (only applications before X date, after Y date, or between X and Y dates)
- have another button that lets me download ONLY the applications visible from the filtering

**Comment by neel-a-mehta** [comment]
When the program admin logs into their Civiform account, they will see a page that lists all the programs they manage (and thus all the forms they are responsible for). When they click into a particular program, they see a button to download all submissions as a CSV. For v1, this can be the only link on the page.

If we’d like to be fancy, we can offer a scrollable list that shows some high-level information about each application in reverse chronological order. It might look like this, minus the checkboxes down the left:



The list would include:

The submission number or ID
The submission time and date 
The applicant’s first name
The applicant’s last name

This list would be purely informational; we wouldn’t show anything if the user clicks on a given row. It’s more to show the program admin the volume of incoming applications at a glance.

Part 2: customizing which CSV rows you export
Once the simple but inefficient strategy in Part 1 is complete, we should give program admins more fine-grained control over which CSV rows they’re exporting. This way, they aren’t re-downloading information dozens of times.

From highest to lowest priority, these allowed customizations are:

Download every application submitted since $DATE 
Download every application submitted from $DATE1 to $DATE2
Download every application submitted since the last time I downloaded
This requires us to track the last time any given program admin hit “Download.”
Download every application submitted before $DATE
Download every application submitted on form version $VERSION

Each download extracts and returns a subset of the entire CSV created in Part 1.

UI/UX
Below the main “Download All” button from Part 1, we should add additional UI widgets to let the user apply these customizations. When they apply a customization, the text on the “Download All” button should change to say “Download X Submissions” or the like, where X is the number of submissions that will be included in the downloaded CSV.

Stretch goal: enable downloading of hand-chosen applications
If we wanted to give the user more fine-grained control, we could let them select individual checkboxes in the list of submissions. When they hit Download, just the selected submissions are included in the CSV.



The user should also be able to use Shift, Command, or Control to select a range of multiple submissions at once.

**Comment by neel-a-mehta** [comment]
An example of what a filterable table could look like: 
![Uploading image.png…]()

**Comment by bxtphung** [comment]
I'm currently working on this after discussion on Thurs 4/22

**Comment by bxtphung** [comment]
Updating with mockup: 
![image](https://user-images.githubusercontent.com/77254850/116939760-84b92300-ac32-11eb-8768-cd855c66413a.png)

Link to Figma: https://www.figma.com/file/CreXQSINm2rzglsIPtGnJW/Program-Admin-View-list-of-applications-and-click-to-view-each

**Comment by neel-a-mehta** [comment]
Looks awesome, Becky! I think you've covered everything we need. Only question is what the mobile view might look like -- do you have a point of view on that?

**Comment by neel-a-mehta** [comment]
A similar mock from Namrataa, btw:

![image](https://user-images.githubusercontent.com/77128518/117355317-6c3e4800-ae80-11eb-83c5-205fa76bf3cd.png)
